### PR TITLE
329 implement notifications

### PR DESCRIPTION
**Pull request by bscheltinga**
Fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/329

### CODE REVIEW

File: Rasa_Bot/data/rules/rules_notifications.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - intent: EXTERNAL_finish_dialog_notification  # TODO: set trigger if user is inactive for 1hour
+  - action: utter_notifications_finish_dialog_1
+  - action: utter_notifications_finish_dialog_2
+
+## Inactive users notification
+- rule: inactive users notification
+  steps:
+  - intent: EXTERNAL_inactive_user_notification  # TODO: set trigger if user inactive for 10 days
+  - action: utter_notifications_user_inactive_1  # TODO: go to utter_central_mode_options
+  - action: utter_central_mode_options
```
Comment (wbaccinelli) [inline Rasa_Bot/data/rules/rules_notifications.yml:32]:
```suggestion
  - action: utter_notifications_user_inactive_1
  - action: action_select_menu
```

We can use `action_select_menu`. It determines whether a dialog has to be completed, and shows the right menu

File: Rasa_Bot/data/rules/rules_notifications.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  - action: action_end_dialog
+
+## New quit date tomorrow notification
+- rule: new quit date tomorrow notification  # TODO: trigger if user set new quit date, day before date
```
Comment (wbaccinelli) [inline Rasa_Bot/data/rules/rules_notifications.yml:36]:
```suggestion
- rule: new quit date tomorrow notification 
```

This is set [here](https://github.com/PerfectFit-project/virtual-coach-rasa/blob/f272bad41b14623845d39a30af2f1593a0f49d75/scheduler/state_machine/controller.py#L465)

File: Rasa_Bot/data/rules/rules_notifications.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+- rule: step goal and intensive pa notification  # TODO: set trigger
+  steps:
+    - intent: EXTERNAL_pa_intensity_notification
+    - action: action_notifications_weekly_intensity_minutes_from_db
+    - action: utter_notifications_step_goal_intensive_pa_1
+    - action: utter_notifications_step_goal_intensive_pa_2
+    - action: utter_notifications_step_goal_intensive_pa_3
+
+## new quit date notification
+- rule: new quit date tomorrow notification  # TODO: trigger if user set new quit date, day of quitting
```
Comment (wbaccinelli) [inline Rasa_Bot/data/rules/rules_notifications.yml:52]:
```suggestion
- rule: new quit date tomorrow notification
```

This is set [here](https://github.com/PerfectFit-project/virtual-coach-rasa/blob/f272bad41b14623845d39a30af2f1593a0f49d75/scheduler/state_machine/controller.py#L465)

**Review by wbaccinelli — APPROVED** [review]
Good job! I left some comments, if you want we can discuss together how to proceed

File: Rasa_Bot/actions/helper.py [hunk 4]
```diff
@@ -858,3 +858,43 @@ def get_faik_text(user_id):
...
+    Args:
+    user_id (int): The user ID for whom the daily step goal is to be retrieved from the database.
+
+    Returns:
+    int: The weekly intensity minutes for the given user, retrieved from the database.
+    """
+
+    # Creat session object to connect db
+    session = get_db_session(db_url=DATABASE_URL)
+    selected = session.query(Users).filter_by(nicedayuid=user_id).one()
```
[the remark this answers is not in the document]
Comment (bscheltinga) [reply Rasa_Bot/actions/helper.py:896]: No, it is set in the weekly reflection: https://github.com/PerfectFit-project/virtual-coach-rasa/blob/a9dad6835d4c3acd17ed6e514f4224b919b26e15/Rasa_Bot/actions/actions_weekly_reflection.py#L145-L168

File: Rasa_Bot/actions/helper.py [hunk 5]
```diff
@@ -858,3 +858,43 @@ def get_faik_text(user_id):
...
+
+    Returns:
+    int: The daily step goal for the given user, retrieved from the database.
+    """
+
+    # Creat session object to connect db
+    session = get_db_session(db_url=DATABASE_URL)
+    selected = session.query(Users).filter_by(nicedayuid=user_id).one()
+    pa_goal = selected.pa_steps_daily_goal
+    session.close()
```
[the remark this answers is not in the document]
Comment (bscheltinga) [reply Rasa_Bot/actions/helper.py:878]: I agree! I think it is not of additional value to add the goal to the db. But I do not know how I can calculate the goal as I don't know how the data wil look like

[the remark this answers is not in the document]
Comment (wbaccinelli) [reply Rasa_Bot/actions/helper.py:878]: Let's use a placeholder function like `pa_goal = compute_pa_goal(user_id)`. The function can be implemented in the helper, and, for now it should return a fixed value. Open then an issue for actually implement this. Does this sound good to you?

**Pull request merged**